### PR TITLE
Improve autogenerated manpages

### LINKED ISSUE #2086: Help2man step creates incorrect titles as part of release process
See review comments https://github.com/zcash/zcash/pull/2085
```
-zcash-cli \- RPC client for the Zcash daemon
+Zcash \- RPC client for the Zcash daemon
```
```
-zcashd \- Network daemon for interacting with the Zcash blockchain
+Zcash \- Network daemon for interacting with the Zcash blockchain
```

### PR DESCRIPTION
**Pull request by str4d**
Cherry-picked from bitcoin/bitcoin#8608.

Closes #2086.

### CODE REVIEW

**Comment by arcalinea** [comment]
This script successfully generates man pages. 

Help2man doesn't recognize "Commands" and "Register Commands" as distinct sections that should be bolded though. This is the same for the upstream autogenerated man pages as well. Maybe this is worth changing the formatting for later.

Screenshot of what I mean: 
zcashtx-man

ACK for this now, though

**Comment by daira** [comment]
Comments are non-blocking. @zkbot r+

File: configure.ac [hunk 1]
```diff
@@ -976,6 +982,7 @@ AC_SUBST(GMPXX_LIBS)
 AC_SUBST(LIBSNARK_LIBS)
 AC_SUBST(LIBZCASH_LIBS)
 AC_CONFIG_FILES([Makefile src/Makefile share/setup.nsi share/qt/Info.plist src/test/buildenv.py])
+AC_CONFIG_FILES([Makefile src/Makefile doc/man/Makefile share/setup.nsi share/qt/Info.plist src/test/buildenv.py])
```
Comment (arcalinea) [inline configure.ac]: Was this duplicated in merge? Think registering src/Makefile twice is causing build.sh to fail

**Comment by arcalinea** [comment]
Build.sh failed locally with the same error as it did on the test server, 
```
+ ./autogen.sh
configure.ac:985: error: `Makefile' is already registered with AC_CONFIG_FILES.
../../lib/autoconf/status.m4:288: AC_CONFIG_FILES is expanded from...
configure.ac:985: the top level
autom4te: /usr/bin/m4 failed with exit status: 1
aclocal: error: echo failed with exit status: 1
autoreconf: aclocal failed with exit status: 1
```

See comment on configure.ac line 985

**Comment by arcalinea** [comment]
@zkbot try

File: configure.ac (hunk 1 above)
Comment (str4d) [inline configure.ac]: This should have been a change, not an addition (there was a merge conflict).

**Comment by str4d** [comment]
@arcalinea thanks for catching that. I'll rebase to merge your change into the original commit, so the cherry-picked commit matches https://github.com/bitcoin/bitcoin/pull/8608/commits/a32c102fb1766301977d093c006f7f6cbf7fbea7 upstream.

**Comment by str4d** [comment]
Done.

@zkbot r+

**Pull request merged**